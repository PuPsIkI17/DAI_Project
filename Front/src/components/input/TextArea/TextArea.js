import React from 'react';
import PropTypes from 'prop-types';
import './TextArea.css'
export default class TextArea extends React.Component {
    render() {
        const options = {};
        this.props.value !== undefined ? options["value"] = this.props.value : options["defaultValue"] = this.props.defaultValue;

        return (
            <div className="input">
                <label htmlFor={this.props.inputId}>{this.props.label}</label>
                <textarea
                    placeholder={this.props.placeholder}
                    name={this.props.inputId}
                    {...options}
                    id={this.props.id}
                    required={this.props.required}
                    onChange={this.props.onChange} />
            </div>
        )
    }
}
TextArea.propTypes = {
    label: PropTypes.string,
    id: PropTypes.string,
    type: PropTypes.string,
    placeholder: PropTypes.string
}
TextArea.defaultProps = {
    label: 'Label',
    id: 'defaultInput',
    type: 'text',
    placeholder: '',
    defaultValue: '',
    required: false,
    onChange: () => { }
}