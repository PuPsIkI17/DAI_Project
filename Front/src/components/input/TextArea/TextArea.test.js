import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import TextArea from './TextArea';
describe('TextArea Component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<TextArea />, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it('renders the UI as expected', () => {
        const input = renderer
            .create(<TextArea />)
            .toJSON();
        expect(input).toMatchSnapshot();
    });
})
