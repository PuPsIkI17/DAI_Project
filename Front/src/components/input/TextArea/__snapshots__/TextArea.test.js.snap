// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea Component renders the UI as expected 1`] = `
<div
  className="input"
>
  <label>
    Label
  </label>
  <textarea
    defaultValue=""
    id="defaultInput"
    onChange={[Function]}
    placeholder=""
    required={false}
  />
</div>
`;
